spec/services/target_matching_service_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe TargetMatchingService do
+  let!(:user) { create(:user) }
+  let(:topic) { target.topic_id }
+  let(:target) do
+    create(:target, user_id: user.id, latitude: 34.902425, longitude: 56.177360, radius: 500)
+  end
+
+  context 'when the radius of each target touches at some point' do
+    context 'when is the same topic and belongs to users' do
+      let!(:target_not_in_radius) do
+        create(:target, topic_id: topic, latitude: 34.905311, longitude: 56.185386, radius: 150)
+      end
+      let!(:target_in_radius) do
+        create(:target, topic_id: topic, latitude: 34.905311, longitude: 56.185386, radius: 900)
+      end
+      let!(:other_target_in_radius) do
+        create(:target, topic_id: topic, latitude: 34.902742, longitude: 56.183395, radius: 100)
+      end
+
+      it 'return a list of matches of my target' do
+        target_matches = TargetMatchingService.new(target).give_me_matches
+        expect(target_matches).to contain_exactly(target_in_radius, other_target_in_radius)
+      end
+    end
+
+    context 'when is not the same topic and belongs to deferents users' do
+      let(:topic) { create(:topic) }
```

Comment:
If you define `topic_a` and `topic_b` on top, you can then omit creating them on this contexts.